src/shipping/pickup-option.ts

```diff
@@ -1,3 +1,49 @@
-export default interface PickupOption {
+interface Item {
+    variant_id: number;
+    quantity: number;
+}
+
+interface Radius {
+    value: number;
+    unit: number;
+}
+
+interface Coordinates {
+    latitude: number;
+    longitude: number;
+}
+
+interface SearchArea {
+    radius: Radius;
+    coordinates: Coordinates;
+}
+
+interface PickupMethod {
+    id: number;
+    location_id: number;
+    display_name: string;
+    collection_instructions: string;
+    collection_time_description: string;
+}
+
+interface Option {
+    pickup_method: PickupMethod;
+    item_quantities: Item;
+}
+
+export interface PickupOptionResult {
+    options: Option[];
+}
+
+export interface PickupOptionRequestBody {
+    search_area: SearchArea;
+    items: Item[];
```

Comment:
I think for the public interface, we should ask the client to pass in cart line item IDs instead, and then we internally map those IDs to product variant IDs.

Reply:
@bc-luke can you please have a look at this

Reply:
I agree this would be friendlier to the developer. It does make our implementation more complex, adding the cart dependency, however. When we go on to make a version of this API with GraphQL, I expect we wouldn't have that dependency as it would go via storefront service direct to the shipping service.

Reply:
Another aspect to consider is the response. We are planning to add the quantities of each item available via a given pickup method in the response, e.g.

```
  {
     "pickupOptions": [
       {
         "pickupMethod": {
           "id": 1,
           "locationId": 2,
           "displayName": "Pickup method 2",
           "collectionInstructions": "Instructions",
           "collectionTimeDescription": "Desc"
         },
         "itemQuantities": [
           {
             "variantId": 1,
             "quantity": 2
           }
         ]
       }
    ]
  }
```

I don't think we can map those back to line item IDs so easily. We may have to pass that extra context through to the service.
```

Reply:
We can have multiple of the same variant in the cart but split across different line items, correct? Perhaps this would help guide toward the sense of using Cart ID. Thoughts?

Reply:
That's correct. We will allow (in the second phase) shoppers to pick up items that may share a variant ID from different locations. The current interface would support this, as you can supply multiple objects with the same variant ID.

An advantage of variant ID though is it does allow developers to query this API before an item is added to the cart if they choose to.

Reply:
Hi @bc-luke, I think there's a bit of misunderstanding. I was asking @animesh1987 to do the mapping on the Checkout SDK level, since it already has a cart loaded in memory and is able to infer the variant ID from it. I wasn't asking to change the API.

I suggested this because clients of Checkout SDK use line item IDs to create consignments. So it'd be more natural to use line item IDs to find available pickup options for the purpose of creating a pickup consignment in checkout.